src/main/java/com/iexec/sms/admin/AdminController.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.iexec.sms.admin;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+@Slf4j
+@RestController("/admin")
+public class AdminController {
+
+    private final ReentrantLock rLock = new ReentrantLock(true);
+
+    private final AdminService adminService;
+
+    public AdminController(AdminService adminService) {
+        this.adminService = adminService;
+    }
+
+    @PostMapping("/backup")
+    public ResponseEntity<String> createBackup() {
+        try {
+            if (tryToAcquireLock()) {
+                return ResponseEntity.ok(this.adminService.createDatabaseBackupFile());
+            } else {
+                return ResponseEntity.status(HttpStatus.TOO_MANY_REQUESTS).build();
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        } finally {
+            if (rLock.isHeldByCurrentThread()) {
+                rLock.unlock();
+            }
+        }
+    }
+
+    @PostMapping("/{storageID}/restore-backup")
+    public ResponseEntity<String> restoreBackup(@PathVariable String storageID, @RequestParam(required = true) String fileName) {
+        try {
+            if (tryToAcquireLock()) {
+                return ResponseEntity.ok(this.adminService.restoreDatabaseFromBackupFile(storageID, fileName));
+            } else {
+                return ResponseEntity.status(HttpStatus.TOO_MANY_REQUESTS).build();
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        } finally {
+            if (rLock.isHeldByCurrentThread()) {
+                rLock.unlock();
+            }
+        }
+    }
```

Comment:
It seems we have a similar pattern for both methods. Should we try and refactor them? Something like the following:
```java
    @PostMapping("/backup")
    public ResponseEntity<String> createBackup() {
        return executeWithLock(this.adminService::createDatabaseBackupFile);
    }

    @PostMapping("/{storageID}/restore-backup")
    public ResponseEntity<String> restoreBackup(@PathVariable String storageID, @RequestParam(required = true) String fileName) {
        return executeWithLock(() -> this.adminService.restoreDatabaseFromBackupFile(storageID, fileName));
    }

    private ResponseEntity<String> executeWithLock(Supplier<String> method) {
        try {
            if (tryToAcquireLock()) {
                return ResponseEntity.ok(method.get());
            } else {
                return ResponseEntity.status(HttpStatus.TOO_MANY_REQUESTS).build();
            }
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
        } finally {
            if (rLock.isHeldByCurrentThread()) {
                rLock.unlock();
            }
        }
    }
```

Reply:
I agree, but I'd rather think about it when the methods are actually implemented.

Reply:
Moreover, 2 other endpoints will be added. Let's keep this refinement for a next PR if possible :pray: